.github/workflows/build-wheel-macos-arm64.yaml

```diff
@@ -83,111 +87,135 @@ jobs:
         ref: ${{ needs.constants.outputs.enzyme_version }}
         path: mlir/Enzyme
 
+      # This step is needed because actions/checkout cannot clone to outside GITHUB_WORKSPACE
+      # https://github.com/actions/checkout/issues/197
+    - name: Copy Submodule to tmp cache directory
```

Comment:
Could you elaborate on this? For instance, why can't the submodules remain in the GitHub workspace, and the build commands be run on that directory?

Reply:
Yes, I simply did not want GITHUB_WORKSPACE to be part of the path for the build. On the self-hosted runner, there are multiple instances of the runner program running, each with a different GITHUB_WORKSPACE. Since that value was part of the cache, if you create the cache on one runner and then restore it on another (as in, the value of GITHUB_WORKSPACE changes), then the build fails.

Though this was not an issue when using GitHub hosted mac, the same issue would've appeared if the value of GITHUB_WORKSPACE changed within that (which can happen).

Reply:
Thanks for elaborating :)